.evergreen/check-rustdoc.sh

```diff
@@ -1,8 +1,23 @@
-#!/bin/sh
+#!/bin/bash
 
 set -o errexit
 
-. ~/.cargo/env
+source ./.evergreen/configure-rust.sh
+
+# docs.rs builds the driver on a read-only file system. to create a more realistic environment, we first
+# build the driver to ensure we have all the deps already in src, and then limit the permissions on that directory
+# and rebuild the docs.
+# this is to help us avoid introducing problems like those described here 
+# https://docs.rs/about/builds#read-only-directories where we or a dependency modify source code during the
+# build process.
+cargo +nightly build
```

Comment:
these might not all be strictly necessary but I just copied the exact invocations that we run `cargo rustdoc` with below for completeness

Reply:
Might be worth putting these into their own little script?

Reply:
I didn't end up doing this since we don't use it anywhere else and the number of scripts we have already feels a little out of hand, but per my comment below I did abstract out the "writing out feature combinations to test" part